dwave/plugins/sklearn/transformers.py

```diff
@@ -0,0 +1,334 @@
+# Copyright 2023 D-Wave Systems Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+import tempfile
+import logging
+import warnings
+
+import dimod
+import numpy as np
+import pandas as pd
+
+from sklearn.base import BaseEstimator
+from sklearn.feature_selection import SelectorMixin
+from dwave.system import LeapHybridCQMSampler
+from dwave.cloud.exceptions import ConfigFileError, SolverAuthenticationError
+from typing import Union
+from .utilities import corrcoef
+
+__all__ = ["SelectFromQuadraticModel"]
+
+
+class SelectFromQuadraticModel(BaseEstimator, SelectorMixin):
+    """scikit-learn `SelectorMizin` which uses the `LeapHybridCQMSampler` solver."""
+
+    acceptable_methods = ["correlation", "mutual information"]
+
+    def __init__(
+        self,
+        alpha: float = 0.5,
+        time_limit: int = 10,
+        n_default_feature: int = 10,
+        method: str = "correlation",
+        chunksize=None,
+    ) -> None:
+        """Instantiate `SelectFromQuadraticModel`
+
+        Args:
+            alpha (float, optional): hyperparameter between 0 and 1, determines how heavily the objective is weighted. Defaults to 0.5.
+            time_limit (int, optional): time limit for the Leap `LeapHybridCQMSampler` solver. Defaults to 10.
+            n_default_feature (int, optional): the number of features to select. This is used if `fit` or `fit_transform` aren't passed `number_of_features`. Defaults to 10.
+            method (str, optional): The method of formulating the feature selection problem, only "correlation" is supported. Defaults to "correlation".
+            chunksize (_type_, optional): Used for testing internal memory usage. Defaults to None.
+        """
```

Comment:
Line length above is excessive: for parameters descriptions you can break the lines at ~100 chars and continue on the next line with an indent, then it renders properly